File: Python/employment.py
import time
from selenium.webdriver.support.ui import WebDriverWait

class Employment(object):
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 30)

    # Selects employment status from dropdown
    def EmploymentStatus(self, status):
        time.sleep(2)
        self.driver.find_elements_by_xpath('//button[@type="button"]')[0].click()
        time.sleep(2)
        self.driver.find_element_by_xpath('//div[text()="' + status + '"]').click()
        time.sleep(2)

    # Inputs Employer Name
    def EmployerName(self, name):
        self.driver.find_element_by_id('employer-name').send_keys(name)

    # Inputs Employee Title
    def EmployeeTitle(self, title):
        self.driver.find_element_by_id('employee-title').send_keys(title)

    # Inputs Employee Start Date
    def EmployeeStartDate(self, start):
        self.driver.find_element_by_id('employee-start-date').send_keys(start)

    # Inputs Employer Phone Number
    def EmployerPhoneNumber(self, phone):
        self.driver.find_element_by_id('employer-phone-number').send_keys(phone)

    # Inputs Gross Monthly Income
    def GrossMonthlyIncome(self, income):
        self.driver.find_element_by_id('gross-monthly-income').send_keys(income)

    # Clicks on button "Next"
    def Navigate(self):
        time.sleep(3)
        self.driver.save_screenshot('Employtment_Information.png')
        self.driver.find_element_by_xpath('//button[text()="Next"]').click()
        self.driver.save_screenshot('Stopping_Point.png')

File: Java/src/PageFactory/Employment.java
package PageFactory;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.concurrent.TimeUnit;

public class Employment
{
    WebDriver driver;
    WebDriverWait wait;

    public Employment(WebDriver driver)
    {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    public void EmploymentStatus(String status) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElements(By.xpath("//button[@type=\"button\"]")).get(0).click();
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.xpath("//div[text()=\"" + status + "\"]")).click();
        TimeUnit.SECONDS.sleep(2);
    }

    public void EmployerName(String name) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.id("employer-name")).sendKeys(name);
    }

    public void EmployeeTitle(String title) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        Actions actions = new Actions(driver);
        WebElement element = driver.findElement(By.id("employee-title"));
        actions.moveToElement(element);
        actions.click();
        actions.sendKeys(title);
        actions.build().perform();
    }

    public void EmployeeStartDate(String start) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.id("employee-start-date")).sendKeys(start);
    }

    public void EmployerPhoneNumber(String phone) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.id("employer-phone-number")).sendKeys(phone);
    }

    public void GrossMonthlyIncome(String income) throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(2);
        driver.findElement(By.id("gross-monthly-income")).sendKeys(income);
    }

    public void Navigate() throws InterruptedException
    {
        TimeUnit.SECONDS.sleep(3);
        driver.findElement(By.xpath("//button[text()=\"Next\"]")).click();
    }
}
